### PR TITLE
Add `promiseToGenerator()` to AwaitStd

### PR DESCRIPTION
**Pull request by Swift-Strider**
Added AwaitStd::promiseToGenerator(), and a new `PromiseRejectedException` to throw when the backing promise is rejected.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	"license": "Apache-2.0",
 	"require": {
 		"pocketmine/pocketmine-mp": "^4.0.0",
-		"sof3/await-generator": "^3.1.1"
+		"sof3/await-generator": "^3.3.0"
```
Comment (Swift-Strider) [inline composer.json:8]: According to [here](https://github.com/SOF3/await-generator/blob/master/CHANGELOG.md#330) v3.3.0 is when await-generator had `Await::promise()` added, which means I had to bump the version in `composer.json`.

File: src/SOFe/AwaitStd/AwaitStd.php [hunk 2]
```diff
@@ -98,4 +100,23 @@ public function timeout(Generator $promise, int $ticks, $onTimeout = null) : Gen
...
+	/**
+	 * Converts a Promise to a Generator by mapping onCompletion and
+	 * onError.
+	 *
+	 * @template V The promise's value
+	 * @throws PromiseRejectedException if the promise gets rejected
+	 * @param Promise<T> $promise
+	 * @return Generator<mixed, mixed, mixed, T>
+	 */
+	public function promiseToGenerator(Promise $promise): Generator {
```
Comment (Swift-Strider) [inline src/SOFe/AwaitStd/AwaitStd.php]: `AwaitStd::promiseToGenerator()` can be marked as static as it doesn't need or use `$this`.

**Review by Swift-Strider — COMMENTED** [review]
I left some comments. Technically, `AwaitStd::promiseToGenerator()` can be a static function. Should I change it to be static?

File: src/SOFe/AwaitStd/AwaitStd.php (hunk 2 above)
Comment (SOF3) [reply src/SOFe/AwaitStd/AwaitStd.php]: yes, please use static

Comment (Swift-Strider) [reply src/SOFe/AwaitStd/AwaitStd.php]: The function is now static

**Comment by Swift-Strider** [comment]
To summarize the changes:
* New **static** function `AwaitStd::promiseToGenerator(Promise $promise): Generator`
* Bumped dependency on "sof3/await-generator" from "^3.1.1" to "^3.3.0"
* New `PromiseRejectedException`

You may want to bump the version of AwaitStd, but I'll leave it up to you to decide.

**Review by SOF3 — APPROVED** [review]

File: src/SOFe/AwaitStd/AwaitStd.php [hunk 3]
```diff
@@ -98,4 +100,23 @@ public function timeout(Generator $promise, int $ticks, $onTimeout = null) : Gen
...
+	 * @template V The promise's value
+	 * @throws PromiseRejectedException if the promise gets rejected
+	 * @param Promise<T> $promise
+	 * @return Generator<mixed, mixed, mixed, T>
+	 */
+	public static function promiseToGenerator(Promise $promise): Generator {
+		return yield from Await::promise(
+			static function($resolve, $reject) use ($promise) {
+				$promise->onCompletion($resolve, static function() use ($reject) {
+					$reject(new PromiseRejectedException('Promise was rejected!'));
```
Comment (SOF3) [inline src/SOFe/AwaitStd/AwaitStd.php:117]: There is no rejection message in Promise?

Comment (Swift-Strider) [reply src/SOFe/AwaitStd/AwaitStd.php:117]: No, it's impossible to reject a promise with a message, [`PromiseResolver::reject()`](https://github.com/pmmp/PocketMine-MP/blob/stable/src/promise/PromiseResolver.php#L57-L67) takes no arguments.

**Review by SOF3 — APPROVED** [review]

**Pull request merged**